### ISSUE TITLE
Find a way to automate testing

### ISSUE THREAD

**Issue by RyanSchw** [PurdueElectricRacing/Wireless-Telemetry-Server#11]
Given the depth of code #7 has as part of feature adding, I think it would be good to revisit the idea of creating some test code for core functionality. I have no clue how this could be done, but I'm curious to see if @AdamBusch or @jacklecroy have any interest or ideas on how to create some test cases.

**Comment by RyanSchw** [comment]
Maybe something like [this](http://abyz.me.uk/rpi/pigpio/python.html) could be useful?

**Comment by jacklecroy** [comment]
Yeah we could try using pytest or something. We would have to manually write tests for everything, though.

**Comment by RyanSchw** [comment]
Sorry, I can better explain. Writing the tests should be a manual process anyways, so I don't mean automatically create tests. I mean that we'll manually write the tests and then use TravisCI (which we use for the linter already) to execute those tests.

The point of difficulty will be testing something like a serial input (so we'll probably have to find mock libraries for everything). Once we figure out how to mock everything, we could use pytest or the native unittest module to test those assertions.

**Comment by jacklecroy** [comment]
OK, is there anything right now you think should have testing?

**Comment by RyanSchw** [comment]
Ideally all of it, lol. Would be good to start off creating tests for the things you know already work (actual physical tests) so that you have a point of reference. Essentially we should be able to verify the system/code will work without having to go outside and run the code on the RPi.


________________________________
